src/components/templates/PartyMap/components/Map/Map.tsx

```diff
@@ -227,32 +245,62 @@ export const Map: React.FC<PropsType> = ({
       )),
     [attendances, setIsRoomModalOpen, setSelectedRoom, venue]
   );
+
+  const zoomIn = useCallback(() => {
+    setZoom((zoom) => Math.min(zoom + ZOOM_INCREMENT, MAX_ZOOM));
+  }, []);
+
+  const zoomOut = useCallback(() => {
+    setZoom((zoom) => Math.max(zoom - ZOOM_INCREMENT, MIN_ZOOM));
+  }, []);
+
   if (!user || !venue) {
     return <>Loading map...</>;
   }
 
+  const translateX =
+    (((mapSize.width * zoom - (mapSize.width * zoom) / zoom) / zoom) * 2) / 4;
+  const translateY =
+    (((mapSize.height * zoom - (mapSize.height * zoom) / zoom) / zoom) * 2) / 4;
+  const bgRatio = mapSize.width ? mapSize.width / mapSize.height : 1;
+
+  const previewAvatarPosition = {
+    x: currentPosition?.row ? (currentPosition?.row / rows) * 100 - 2 : 0,
+    y: currentPosition?.column
+      ? (currentPosition?.column / columns) * 100 - 2
+      : 0,
+  };
+
+  const zoomableMapSize = showGrid
+    ? {
+        width: mapSize.width,
+        height: mapSize.height,
+      }
+    : {};
+
   return (
     <div className="party-map-content-container">
       <div className="party-map-container">
-        <div className="party-map-content">
-          <img
-            width="100%"
-            className="party-map-background"
-            src={venue.mapBackgroundImageUrl}
-            alt=""
-          />
-
+        <div
+          className="party-map-content"
+          style={{
+            ...zoomableMapSize,
+            transform: `scale(${zoom}) translate3d(${translateX}px, ${translateY}px, 0)`,
+          }}
+        >
           <div
             className="party-map-grid-container"
             style={{
+              ...zoomableMapSize,
+              backgroundImage: `url(${venue.mapBackgroundImageUrl})`,
```

Comment:
Do we need to spread `zoomableMapSize` in both of these locations? if so, why is that?